# Finding the Confidence Interval of Polling Figures

You are running a political campaign and decide to run 30 focus groups with about 10 people in each group. You get the results and want to report to your candidate the number of people who would vote for them in a typical 10-person group. Since there is some variability in each focus group, you decide that the most accurate way is to give a 95% z-confidence interval. You assume from past experience that the standard deviation is 2.89.

1.Import the random Python package and set the seed to 39809. This will ensure that we get the same results every time we run the program:

In [10]:
import random
import numpy as np
import scipy.stats as stats
import math
random.seed(39809)

2.Initialize our sample list and collect our samples from our focus groups. Use random.randint

In [11]:
sample_list = []
for i in range(30):
    sample_list.append(random.randint(0, 10))
    
sample_mean = np.mean(sample_list)

3.Calculate 95% z-confidence interval.

In [13]:
ci = stats.norm.interval(0.95, sample_mean, 2.89/math.sqrt(len(sample_list)))
ci

(3.965845784931483, 6.034154215068517)

4.If you did everything correctly, then the following should be printed when you run your notebook:

    Your 0.95 z confidence interval is (3.965845784931483, 6.034154215068517)

# Hypothesis Testing

Your boss asks you to conduct a hypothesis test about the mean dwell time of a new type of UAV.  Before you arrived, an experiment was conducted on n=5 UAVs (all of the new type) resulting in a sample mean dwell time of ybar=10.4 hours.  The goal is to conclusively demonstrate, if possible, that the data supports the manufacturer’s claim that the mean dwell time is greater than 10 hours.  Given that it is reasonable to assume the dwell times are normally distributed, the sample standard deviation is s = 0.5 hours, and using a significance level of α = 0.01:

1.Write out the null and alternative hypotheses

In [ ]:
The null and alternative hypotheses: H0: μ = 10, Ha: μ > 10

2.Calculate the test statistic

In [16]:
mu = 10
sample_mean = 10.4
sample_st_dev = 0.5
sample_size = 5
sig_level = 0.01

test_statistic = (sample_mean - mu) / (sample_st_dev/(math.sqrt(sample_size)))
test_statistic

1.7888543819998335

3.Find the p-value and state the outcome

In [17]:
p_value = 1 - stats.t.cdf(test_statistic, df=sample_size-1)

print(f'P Value = {p_value}')
if p_value < sig_level:
    print('Results are significant. Reject the Null')
else:
    print('Do Not Reject the Null')

P Value = 0.07407407407407385
Do Not Reject the Null
